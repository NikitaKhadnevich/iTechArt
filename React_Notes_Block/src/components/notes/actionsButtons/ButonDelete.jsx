/* eslint-disable import/no-cycle */
import React from 'react';
import PropTypes from 'prop-types';
import Stack from '@mui/material/Stack';
import IconButton from '@mui/material/IconButton';
import DeleteIcon from '@mui/icons-material/Delete';

const ButtonDelete = ({ index, handleDelete, id }) => (
  <>
    <Stack direction='row' spacing={0.5} alignItems='center'>
      <IconButton
        aria-label='delete'
        color='primary'
        onClick={() => handleDelete(index, id)}
      >
        <DeleteIcon />
      </IconButton>
    </Stack>
  </>
);

ButtonDelete.propTypes = {
  handleDelete: PropTypes.func,
  index: PropTypes.number,
  id: PropTypes.number,
};

ButtonDelete.defaultProps = {
  handleDelete: 'handleDelete',
  index: 'index',
  id: 'id',
};

export default ButtonDelete;
