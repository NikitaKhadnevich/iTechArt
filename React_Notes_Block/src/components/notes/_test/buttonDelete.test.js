/* eslint-disable no-unused-vars */
/* eslint-disable no-shadow */
/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import * as reactRedux from 'react-redux';
import { configure, shallow, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import ButonDelete from '../actionsButtons/ButonDelete';

configure({ adapter: new Adapter() }); // ES6 modules

describe('ButonDelete', () => {
  const butonDeleteProps = {
    index: 1,
    id: 2,
    handleDelete: jest.fn(
      (index, id) => `call delete with index: ${index} and id: ${id}`
    ),
  };

  it('Is it ButonDelete Exist', () => {
    const buttonDelete = shallow(<ButonDelete {...butonDeleteProps} />);
    expect(buttonDelete).toMatchSnapshot();
    expect(buttonDelete).toHaveLength(1);
  });
  it('Test Delete Event', () => {
    const buttonDelete = shallow(<ButonDelete {...butonDeleteProps} />);
    const { index, id, handleDelete } = butonDeleteProps;
    buttonDelete.find('ForwardRef(IconButton)').simulate('click');
    expect(handleDelete).toHaveBeenCalledTimes(1);
    expect(handleDelete).toHaveBeenCalledWith(index, id);
  });
});
