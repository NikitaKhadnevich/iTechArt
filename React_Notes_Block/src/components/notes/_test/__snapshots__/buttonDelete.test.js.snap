// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButonDelete Is it ButonDelete Exist 1`] = `
<Fragment>
  <ForwardRef(Stack)
    alignItems="center"
    direction="row"
    spacing={0.5}
  >
    <ForwardRef(IconButton)
      aria-label="delete"
      color="primary"
      onClick={[Function]}
    >
      <Memo(ForwardRef(DeleteIcon)) />
    </ForwardRef(IconButton)>
  </ForwardRef(Stack)>
</Fragment>
`;
